src/context.rs

```diff
@@ -1360,30 +1360,6 @@ impl BlockContext {
     &self.blocks[bo.y][bo.x]
   }
 
-  pub fn above_of(&self, bo: BlockOffset) -> Block {
```

Comment:
You may change it to return an Option<&Block> and have the code use `if let` or `map().unwrap_or()` instead.

Reply:
Yes, but this would only avoid to return `Block` by value.

The boundaries checks are (often) redundant (already done by the callers), so I would like to also remove them. When they are already checked by the callers, it would be annoying to (also) unwrap the `Option`.

IMO, these helpers functions do not really help compared to direct indexing.